src/main/java/com/amashchenko/maven/plugin/gitflow/AbstractGitFlowMojo.java

```diff
@@ -1288,5 +1362,196 @@ public String getError() {
+        }
+
+        final boolean noChangelistValueToBeModified = productionChangelistValue == null
+                && hotfixChangelistValue == null && releaseChangelistValue == null
+                && developmentChangelistValue == null && featureChangelistValue == null
+                && supportChangelistValue == null;
+
+        if (StringUtils.isBlank(changelistProperty) || noChangelistValueToBeModified) {
+            return;
+        }
+
+        switch (branchType) {
+            case PRODUCTION:
+                setChangelistPropertyToValue(productionChangelistValue);
+                break;
+            case HOTFIX:
+                setChangelistPropertyToValue(hotfixChangelistValue);
+                break;
+            case RELEASE:
+                setChangelistPropertyToValue(releaseChangelistValue);
+                break;
+            case DEVELOPMENT:
+                setChangelistPropertyToValue(developmentChangelistValue);
+                break;
+            case FEATURE:
+                setChangelistPropertyToValue(featureChangelistValue);
+                break;
+            case SUPPORT:
+                setChangelistPropertyToValue(supportChangelistValue);
+                break;
+        }
+    }
+
+    /**
+     * Sets the <code>changelist</code> property globally.
+     * 
+     * @param changelistPropertyValue
+     *            The value to set the <code>changelist</code> property to.
+     *            Remove the property if the value is null.
+     */
+    private void setChangelistPropertyToValue(final String changelistPropertyValue) {
+        setProperty(changelistProperty, changelistPropertyValue, mavenSession);
+    }
+
+    /**
+     * Sets a property globally in system-properties, optionally in the properties of the
+     * <code>MavenSession</code> and replaces the argLine to contain the value as well.
+     * 
+     * @param key
+     *            The key of the property to set.
+     * @param value
+     *            The value of the property to set, if null, the property gets removed.
+     * @param session
+     *            An optional <code>MavenSession</code> can be passed, to replace the properties
+     *            in it's property-lists.
+     */
+    private void setProperty(final String key, final String value, final MavenSession session) {
+        if (StringUtils.isBlank(key)) {
+            return;
+        }
+
+        setPropertyInProperties(key, value, System.getProperties());
+        if (session != null) {
+            setPropertyInProperties(key, value, session.getUserProperties());
+            setPropertyInProperties(key, value, session.getSystemProperties());
+            setPropertyInProperties(key, value, session.getProjectBuildingRequest().getUserProperties());
+            setPropertyInProperties(key, value, session.getProjectBuildingRequest().getSystemProperties());
+        }
+
+        argLine = replacePropertyInArgline(key, value, argLineOrig);
```

Comment:
Of course it is needed, so that the info is passed to the maven calls. But it may be more beautiful to append args in `executeMavenCommand` instead of manipulating the `argLine`.